kyaml/yaml/fns.go

```diff
@@ -273,19 +343,28 @@ func (e ElementMatcher) Filter(rn *RNode) (*RNode, error) {
 		var err error
 		if e.MatchAnyValue {
 			field, err = elem.Pipe(Get(e.FieldName))
+			if IsFoundOrError(field, err) {
+				return elem, err
+			}
 		} else {
-			field, err = elem.Pipe(MatchField(e.FieldName, e.FieldValue))
-		}
-		if IsFoundOrError(field, err) {
-			return elem, err
+			found := true
+			for i, key := range e.Keys {
+				field, err = elem.Pipe(MatchField(key, e.Values[i]))
+				if !IsFoundOrError(field, err) {
+					found = false
+					break
+				}
+			}
```

Comment:
I'm not quite familiar with this code snippet. The logic looks a little bit awkward. We normally start `found` with false, and break once "found" a match. I guess it may be a naming issue. how about changing "found" to "isMissing". So as the code can be read as "If one key is not found, isMissing is true."